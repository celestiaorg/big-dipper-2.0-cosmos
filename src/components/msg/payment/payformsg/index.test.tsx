import React from 'react';
import { RecoilRoot } from 'recoil';
import renderer from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import { MsgPayForMsg } from '@models';
import Send from '.';

// ==================================
// mocks
// ==================================

jest.mock('@components', () => ({
  Name: (props) => <div id="Name" {...props} />,
}));

// ==================================
// unit tests
// ==================================
describe('screen: TransactionDetails/MsgPayForMsg', () => {
  it('matches snapshot', () => {
    const message = new MsgPayForMsg({
      category: 'payment',
      type: 'MsgPayForMsg',
      signer: 'signer',
      message_namespace_id: 'message_namespace_id',
      message_size: 'message_size',
      message_share_commitment: 'message_share_commitment',
    });
    const component = renderer.create(
      <RecoilRoot>
        <MockTheme>
          <Send
            message={message}
          />
        </MockTheme>
      </RecoilRoot>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
