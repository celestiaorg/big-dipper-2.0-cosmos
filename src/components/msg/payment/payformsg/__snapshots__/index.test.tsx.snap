// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: TransactionDetails/MsgPayForMsg matches snapshot 1`] = `
<p
  className="MuiTypography-root MuiTypography-body1"
>
  message_contents:txPayForMsgContent
</p>
`;
